packages/fx-core/src/core/question.ts

```diff
@@ -495,23 +497,34 @@ export const QuestionSelectResourceGroup: SingleSelectQuestion = {
   skipSingleOption: true,
   forgetLastValue: true,
 };
-
-export const QuestionNewResourceGroupName: TextInputQuestion = {
-  type: "text",
-  name: CoreQuestionNames.NewResourceGroupName,
-  title: getLocalizedString("core.QuestionNewResourceGroupName.title"),
-  validation: {
+export function newResourceGroupNameQuestion(
+  rmClient: ResourceManagementClient
+): TextInputQuestion {
+  const question = QuestionNewResourceGroupName;
+  question.validation = {
     validFunc: async (input: string): Promise<string | undefined> => {
       const name = input as string;
       // https://docs.microsoft.com/en-us/rest/api/resources/resource-groups/create-or-update#uri-parameters
       const match = name.match(/^[-\w._()]+$/);
       if (!match) {
         return getLocalizedString("core.QuestionNewResourceGroupName.validation");
       }
-
+      const maybeExist = await resourceGroupHelper.checkResourceGroupExistence(name, rmClient);
+      if (maybeExist.isErr()) {
+        return maybeExist.error.message;
+      }
+      if (maybeExist.value) {
+        return `resource group already exists: ${name}`;
```

Comment:
Do we need to localize this string?

Reply:
We can add later in a unified manner. Current code logic has no localization:
<img width="665" alt="image" src="https://user-images.githubusercontent.com/1658418/178233671-a557b9f6-7c6e-4c4f-b43b-b3bf4e3a69c5.png">